core/src/main/java/org/apache/iceberg/data/avro/DataReader.java

```diff
@@ -19,192 +19,25 @@
 
 package org.apache.iceberg.data.avro;
 
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
-import org.apache.avro.LogicalType;
-import org.apache.avro.LogicalTypes;
-import org.apache.avro.Schema;
-import org.apache.avro.io.DatumReader;
-import org.apache.avro.io.Decoder;
-import org.apache.avro.io.DecoderFactory;
-import org.apache.avro.io.ResolvingDecoder;
-import org.apache.iceberg.avro.AvroSchemaUtil;
-import org.apache.iceberg.avro.AvroSchemaWithTypeVisitor;
-import org.apache.iceberg.avro.ValueReader;
-import org.apache.iceberg.avro.ValueReaders;
-import org.apache.iceberg.exceptions.RuntimeIOException;
+import org.apache.iceberg.Schema;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
-import org.apache.iceberg.relocated.com.google.common.collect.MapMaker;
-import org.apache.iceberg.types.Type;
-import org.apache.iceberg.types.Types;
 
-public class DataReader<T> implements DatumReader<T> {
-
-  private static final ThreadLocal<Map<Schema, Map<Schema, ResolvingDecoder>>> DECODER_CACHES =
-      ThreadLocal.withInitial(() -> new MapMaker().weakKeys().makeMap());
+/**
+ * @deprecated will be removed, please use AvroGenericReader instead.
+ */
+@Deprecated
+public class DataReader<T> extends BaseAvroGenericReader<T> {
+  DataReader(Schema expectedSchema, org.apache.avro.Schema readSchema, Map<Integer, ?> idToConstant) {
+    super(expectedSchema, readSchema, idToConstant);
+  }
 
-  public static <D> DataReader<D> create(org.apache.iceberg.Schema expectedSchema, Schema readSchema) {
+  public static <D> DataReader<D> create(Schema expectedSchema, org.apache.avro.Schema readSchema) {
```

Comment:
I think this changes the semantics of deprecated `DataReader#create`,  the arguments has been swapped.

Reply:
No, I import iceberg schema instead of avro schema. Let me update this to make it clear.